crates/ra_ide_api/src/completion/complete_dot.rs

```diff
@@ -1,19 +1,32 @@
 use hir::{AdtDef, Ty, TypeCtor};
 
-use crate::completion::{CompletionContext, Completions};
+use crate::completion::completion_item::CompletionKind;
+use crate::{
+    completion::{completion_context::CompletionContext, completion_item::Completions},
+    CompletionItem,
+};
 use rustc_hash::FxHashSet;
 
-/// Complete dot accesses, i.e. fields or methods (currently only fields).
+/// Complete dot accesses, i.e. fields or methods (and .await syntax).
 pub(super) fn complete_dot(acc: &mut Completions, ctx: &CompletionContext) {
-    let receiver_ty =
-        match ctx.dot_receiver.as_ref().and_then(|it| ctx.analyzer.type_of(ctx.db, it)) {
-            Some(it) => it,
-            None => return,
```

Comment:
Let's keep this early return pattern? 

I personally find it's more readable, because the indent level is smaller and there's less things to keep in the mental stack